pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/service/admin/UserAdminTestUtil.java

```diff
@@ -0,0 +1,404 @@
+        int roleId = getRoleId(role);
+        UserAdminService.deleteUserRole(userId, roleId, studyGroupId);
+        removeRoleForUser(roleId, userId);
+    }
+
+    protected void removeRoleForUser(int roleId, int userId) {
+        List<Integer> roleIds = createdUserRoles.get(userId);
+        if (roleIds == null) {
+            throw new DsmInternalError("Invalid user: " + userId);
+        }
+        List<Integer> newRoleIds = new ArrayList<>(roleIds);
+        newRoleIds.removeAll(List.of(roleId));
+        createdUserRoles.put(userId, newRoleIds);
+    }
+
+    protected void removeRoleForUser(String role, int userId) {
+        removeRoleForUser(getRoleId(role), userId);
+    }
+
+    public void verifyUserHasRole(int userId, String role) {
+        assertInitialized();
+        int userRoleId = UserAdminService.getUserRole(userId, getRoleId(role), studyGroupId);
+        Assert.assertNotEquals(-1, userRoleId);
+    }
+
+    public int getRoleId(String roleName) {
+        assertInitialized();
+        Integer roleId = allRoles.get(roleName);
+        if (roleId == null) {
+            throw new DsmInternalError("Invalid role name: " + roleName);
+        }
+        return roleId;
+    }
+
+    protected static Map<String, Integer> getAllRoles() {
+        return inTransaction(conn -> {
+            Map<String, Integer> roles = new HashMap<>();
+            try (PreparedStatement stmt = conn.prepareStatement("SELECT ar.role_id, ar.name FROM access_role ar")) {
+                try (ResultSet rs = stmt.executeQuery()) {
+                    while (rs.next()) {
+                        roles.put(rs.getString(2), rs.getInt(1));
+                    }
+                }
+            } catch (SQLException e) {
+                throw new DsmInternalError("Error getting roles", e);
+            }
+            return roles;
+        });
+    }
+
+    private static int createTestInstance(String instanceName, int studyGroupId) {
+        int instanceId = createInstance(instanceName);
+        SimpleResult res = inTransaction(conn -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_INSERT_DDP_INSTANCE_GROUP, Statement.RETURN_GENERATED_KEYS)) {
+                stmt.setInt(1, instanceId);
+                stmt.setInt(2, studyGroupId);
+                int result = stmt.executeUpdate();
+                if (result != 1) {
+                    dbVals.resultException = new DsmInternalError("Result count was " + result);
+                }
+                try (ResultSet rs = stmt.getGeneratedKeys()) {
+                    if (rs.next()) {
+                        dbVals.resultValue = rs.getInt(1);
+                    }
+                }
+            } catch (SQLException ex) {
+                dbVals.resultException = ex;
+            }
+            return dbVals;
+        });
+
+        if (res.resultException != null) {
+            _deleteInstance(instanceId);
```

Comment:
Maybe wrap `_deleteInstance` in a try/catch to preserve the originating `res.resultException` error?